packages/xo-server/src/xapi/mixins/patching.mjs

```diff
@@ -629,7 +630,7 @@ export default {
         continue
       }
 
-      const residentVms = await this.getField('host', host.$ref,'resident_VMs')
+      const residentVms = await asyncMap(await this.getField('host', host.$ref,'resident_VMs'), ref => this.getField('VM', ref, 'uuid'))
```

Comment:
Or simply use refs in `vmsByHost`, it will make the code simpler.

(But the var name should reflect that, there is currently much confusion in this code between `$id` and `uuid`)